File: app/src/main/java/com/lokate/app/api/ApiService.java
package com.lokate.app.api;

import com.lokate.app.models.Group;
import com.lokate.app.models.LoginRequest;
import com.lokate.app.models.LoginResponse;
import com.lokate.app.models.OtpRequest;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;

public interface ApiService {
    
    @POST("sendotp/")
    Call<Void> sendOtp(@Body OtpRequest request);
    
    @POST("verifyotp/")
    Call<LoginResponse> verifyOtp(@Body LoginRequest request);
    
    @GET("groups/")
    Call<List<Group>> getGroups(@Header("Authorization") String token);
}

File: app/src/main/java/com/mobiletracker/app/data/api/ApiService.kt
package com.mobiletracker.app.data.api

import com.mobiletracker.app.data.models.*
import retrofit2.Response
import retrofit2.http.*

interface ApiService {

    @POST("api/v1/send-otp/")
    suspend fun sendOtp(@Body otpUser: OtpUser): Response<ApiResponse<String>>

    @POST("api/v1/signin/")
    suspend fun signin(@Body signinUser: SigninUser): Response<RefreshTokenResponse>

    @POST("api/v1/refresh-token/")
    suspend fun refreshToken(@Body request: RefreshTokenRequest): Response<RefreshTokenResponse>

    @GET("api/v1/csrf-token")
    suspend fun getCsrfToken(): Response<Map<String, String>>

    @POST("api/v1/profiles/")
    suspend fun createProfile(
        @Header("Authorization") token: String,
        @Body profile: Profile
    ): Response<Profile>

    @GET("api/v1/profiles/")
    suspend fun getProfile(
        @Header("Authorization") token: String
    ): Response<Profile>

    @PUT("api/v1/profiles/")
    suspend fun updateProfile(
        @Header("Authorization") token: String,
        @Body profile: Profile
    ): Response<Profile>

    @POST("api/v1/groups/")
    suspend fun createGroup(
        @Header("Authorization") token: String,
        @Body group: Map<String, Any>
    ): Response<Group>

    @GET("api/v1/groups/admin/")
    suspend fun getGroups(
        @Header("Authorization") token: String
    ): Response<List<Group>>

    @GET("api/v1/groups/{group_id}")
    suspend fun getGroup(
        @Header("Authorization") token: String,
        @Path("group_id") groupId: Int
    ): Response<Group>

    @GET("api/v1/group-members/{group_id}")
    suspend fun getGroupMembers(
        @Header("Authorization") token: String,
        @Path("group_id") groupId: Int
    ): Response<List<GroupMember>>

    @PUT("api/v1/group-members/{member_id}/location")
    suspend fun updateLocation(
        @Header("Authorization") token: String,
        @Path("member_id") memberId: Int,
        @Body location: Map<String, Double>
    ): Response<GroupMember>
}